### PR TITLE
Improve parsing of assembly qualified name

### PR DESCRIPTION
**Pull request by iyaneruz0**
- Simplified parsing behaviour.
- Corrected and expanded unit tests to check types are loaded.

### CODE REVIEW

File: tests/Silverback.Core.Tests/Util/TypesCacheTests.cs [hunk 1]
```diff
@@ -9,7 +9,10 @@
 
 namespace Silverback.Tests.Core.Util
 {
-    public class TypesCacheTests
+    /// <summary>
+    /// The types cache unit tests.
+    /// </summary>
```
Comment (BEagle1984) [inline tests/Silverback.Core.Tests/Util/TypesCacheTests.cs:14]: Unecessary summary, you can remove it.

File: tests/Silverback.Core.Tests/Util/TypesCacheTests.TestData.cs [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+                     typeof(List<char>)),
+            };
+        }
+
+        [System.Diagnostics.CodeAnalysis.SuppressMessage("Design", "CA1034:Nested types should not be visible", Justification = "Record belongs to unit test data.")]
+        public record AssemblyQualifiedGenericTypeResult(string CleanAssemblyQualifiedName, Type MatchingType)
+        {
+        }
+
+        /// <summary>
```
Comment (BEagle1984) [inline tests/Silverback.Core.Tests/Util/TypesCacheTests.TestData.cs:86]: Unecessary summary, you can remove it.

File: tests/Silverback.Core.Tests/Util/TypesCacheTests.TestData.cs [hunk 3]
```diff
@@ -0,0 +1,100 @@
+﻿// Copyright (c) 2020 Sergio Aquilini
+// This code is licensed under MIT license (see LICENSE file for details)
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+
+namespace Silverback.Tests.Core.Util
+{
+    /// <summary>
```
Comment (BEagle1984) [inline tests/Silverback.Core.Tests/Util/TypesCacheTests.TestData.cs:10]: Unecessary summary, you can remove it.

**Review by BEagle1984 — CHANGES_REQUESTED** [review]
Looks good but modify the test to ensure that the `Type` is actually resolved.
 
Please rebase and squash your changes into a single commit. The commit message should follow the [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/) guideline.

File: tests/Silverback.Core.Tests/Util/TypesCacheTests.cs (hunk 1 above)
Comment (iyaneruz0) [reply tests/Silverback.Core.Tests/Util/TypesCacheTests.cs]: Removed

File: tests/Silverback.Core.Tests/Util/TypesCacheTests.TestData.cs (hunk 3 above)
Comment (iyaneruz0) [reply tests/Silverback.Core.Tests/Util/TypesCacheTests.TestData.cs]: Removed

**Comment by iyaneruz0** [comment]
I've rebased and changed the commit message, do tell if the message is clear enough.

File: tests/Silverback.Core.Tests/Util/TypesCacheTests.TestData.cs (hunk 2 above)
Comment (BEagle1984) [reply tests/Silverback.Core.Tests/Util/TypesCacheTests.TestData.cs:76]: It's still there though. 😆

Comment (iyaneruz0) [reply tests/Silverback.Core.Tests/Util/TypesCacheTests.TestData.cs]: Sorry, I'm dumb, confused it with the other summary 🤣

**Review by BEagle1984 — APPROVED** [review]

**Pull request merged**